src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,318 @@
+                    throw new InvalidOperationException($"No event stream returned for field '{node.Field.Name}'.");
+                }
+            }
+            catch (OperationCanceledException) when (context.CancellationToken.IsCancellationRequested)
+            {
+                throw;
+            }
+            catch (ExecutionError error)
+            {
+                error.Path = node.ResponsePath;
+                error.AddLocation(node.Field, context.Document);
+                context.Errors.Add(error);
+                return null;
+            }
+            catch (Exception exception)
+            {
+                context.Errors.Add(await HandleExceptionInternalAsync(context, node, exception,
+                    $"Could not subscribe to field '{node.Field.Name}'.").ConfigureAwait(false));
+                return null;
+            }
+
+            // cannot throw an exception here
+            return subscription
+                .SelectCatchAsync(
+                    async (value, token) =>
+                    {
+                        // duplicate context to prevent multiple event streams from sharing the same context,
+                        // and clear errors/metrics/extensions. Free array pool leased arrays after execution.
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
+
+                        return await ProcessDataAsync(childContext, node, value).ConfigureAwait(false);
+                    },
+                    async (exception, token) =>
+                    {
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
+
+                        return await ProcessErrorAsync(context, node, exception).ConfigureAwait(false);
+                    });
+        }
+
+        /// <summary>
+        /// Processes data from the event source via <see cref="IObserver{T}.OnNext(T)"/> and
+        /// returns an <see cref="ExecutionResult"/>.
+        /// <br/><br/>
+        /// Override this method to mutate <see cref="ExecutionContext"/> as necessary, such
+        /// as changing the <see cref="ExecutionContext.RequestServices"/> property to a scoped instance.
+        /// </summary>
+        protected virtual async Task<ExecutionResult> ProcessDataAsync(ExecutionContext context, ExecutionNode node, object? value)
```

Comment:
What if `value` is `null`. "Normal" document executer does not execute request tree further in such case.

Reply:
I have tests for such a case -- see `Widget_Null` and `Widget_Null_Nullable`.